src/Auth/Actions/articleActions.js

```diff
@@ -0,0 +1,90 @@
+import { apiRequest } from '../../Utils/axios';
+import { toast } from 'react-toastify';
+
+import * as types from './types';
+import { getFeedDetails } from './feedActions';
+
+export const getCategoryRequest = () => {
+    return {
+        type: types.GET_CATEGORIES_REQUEST,
+    };
+};
+
+export const getCategorySuccess = (success) => {
+    return {
+        type: types.GET_CATEGORIES_SUCCESS,
+        payload: success,
+    };
+};
+
+export const getCategoryFailure = (error) => {
+    return {
+        type: types.GET_CATEGORIES_FAILURE,
+        payload: error,
+    };
+};
+export const PostArticlesRequest = () => {
+    return {
+        type: types.POST_ARTICLES_REQUEST,
+    };
+};
+
+export const PostArticlesSuccess = (success) => {
+    return {
+        type: types.POST_ARTICLES_SUCCESS,
+        payload: success,
+    };
+};
+
+export const PostArticlesFailure = (error) => {
+    return {
+        type: types.POST_ARTICLES_FAILURE,
+        payload: error,
+    };
+};
+export function getCategory() {
+    return (dispatch) => {
+        const promise = apiRequest('GET', `v1/categories`);
+        dispatch(getCategoryRequest());
+        promise.then(
+            function (payload) {
+                const categories = payload.data;
+                if (categories.status === 'Failed') {
```

Comment:
why are you checking this status if we have a catch block where we are handling errors?